src/flexmeasures_client/response_handling.py

```diff
@@ -0,0 +1,51 @@
+
+from aiohttp import ContentTypeError
+import asyncio
+
+
+async def check_response(self, response):
+    """
+    <300: passes
+    401: reauthenticate
+    todo: 503 + Retry-After header: poll again
+    otherwise: call error_handler
+    """
+    status = response.status
+    payload = await response.json()
+    headers = response.headers
+    if status < 300:
+        pass
+    elif status == 401:
+        self.get_access_token()
+        self.reauth_once = False
+    elif status == 503 and "Retry-After" in headers:
+        # todo: move the client_should_retry logic into this function)
+        pass
+    elif status == 400 and (
+        "Scheduling job waiting" in payload.get("message", "")
+        or "Scheduling job in progress" in payload.get("message", "")
+    ):
+        print(
+            f"Server indicated to try again later. Retrying in {self.polling_interval} seconds..."
+        )
+        self.polling_step += 1
+        await asyncio.sleep(self.polling_interval)
+    else:
+        response.raise_for_status
```

Comment:
shouldn't this be called, i.e `response.raise_for_status()`?

Reply:
I'll write a test

Reply:
Test written.